src/modules/commander/PreflightCheck.cpp

```diff
@@ -602,13 +644,16 @@ bool preflightCheck(orb_advert_t *mavlink_log_pub, bool checkSensors, bool check
 		int32_t prime_id = 0;
 		param_get(param_find("CAL_MAG_PRIME"), &prime_id);
 
+		bool mag_fail_reported = false;
+
 		/* check all sensors, but fail only for mandatory ones */
 		for (unsigned i = 0; i < max_optional_mag_count; i++) {
 			bool required = (i < max_mandatory_mag_count);
 			int device_id = -1;
 
-			if (!magnometerCheck(mavlink_log_pub, i, !required, device_id, reportFailures) && required) {
+			if (!magnometerCheck(mavlink_log_pub, i, !required, device_id, (reportFailures && !mag_fail_reported)) && required) {
 				failed = true;
+				mag_fail_reported = true;
```

Comment:
The `mag_fail_reported` flag doesn't do anything as I understand it. Since `max_mandatory_mag_count` is 1, only the first check is required. After that `(reportFailures && !mag_fail_reported)) && required` evaluates to `false` anyway.

Reply:
It should prevent the repetition of 

    PREFLIGHT FAIL: MAG 0 UNCALIBRATED
    PREFLIGHT FAIL: MAG 1 UNCALIBRATED
    PREFLIGHT FAIL: MAG 2 UNCALIBRATED

Since you can only calibrate them in bulk, I didn't see the point in reporting more than a single line.

Reply:
But if you only report for mandatory devices then you don't need the reported flag, `required` should be enough.

Reply:
It still reports uncalibrated for optional devices. I find the entire thing more convoluted than necessary. It could almost be refactored into a template for all types.